tests/unit/domain/validation/test_validation_new_features.py

```
@@ -60,6 +60,21 @@ def test_template_lints_vt2_vt3_vt4() -> None:
             for w in report.warnings:
                 print(f"  {w.rule_id}: {w.message}")
 
+            # Additional debugging: check the actual step structure
+            print("DEBUG: Pipeline steps:")
+            for idx, step in enumerate(pipeline.steps):
+                print(f"  Step {idx}: name={getattr(step, 'name', None)}")
+                meta = getattr(step, "meta", {})
+                templ = meta.get("templated_input", None)
+                print(f"    templated_input: {repr(templ)}")
```

Ruff (0.14.5)

Use explicit conversion flag

Replace with conversion flag

(RUF010)